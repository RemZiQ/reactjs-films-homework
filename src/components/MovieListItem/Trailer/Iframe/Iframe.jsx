import React from 'react';
import propTypes from 'prop-types';

const Iframe = ({ trailer, error }) => {
  if (error === true) {
    throw new Error('Here is no trailer');
  }
  return (
    <iframe title="trailer" src={trailer} width="640" height="480" frameBorder="0" allowFullScreen />
  );
};

Iframe.propTypes = {
  trailer: propTypes.string,
  error: propTypes.bool,
};

Iframe.defaultProps = {
  trailer: null,
  error: null,
};

export default Iframe;
