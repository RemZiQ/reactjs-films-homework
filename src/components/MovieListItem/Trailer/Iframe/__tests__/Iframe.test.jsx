import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { Iframe } from '../Iframe';


it('Iframe renders well', () => {
  const props = {
    trailer: 'someUrl',
  };
  const renderer = new ShallowRenderer();
  const result = renderer.render(<Iframe {...props} />);
  expect(result).toMatchSnapshot();
});

it('Iframe throw', () => {
  const props = {
    error: true,
  };
  const renderer = new ShallowRenderer();
  expect(() => renderer.render(<Iframe {...props} />)).toThrowError();
});
