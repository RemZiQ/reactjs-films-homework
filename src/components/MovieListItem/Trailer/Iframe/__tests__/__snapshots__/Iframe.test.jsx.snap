// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Iframe renders well 1`] = `
<iframe
  allowFullScreen={true}
  frameBorder="0"
  height="480"
  src="someUrl"
  title="trailer"
  width="640"
/>
`;
